src/compressor/Compressor.h

```diff
@@ -103,4 +139,15 @@ class Compressor {
 
 };
 
+struct CompConnectionMeta {
+  uint32_t con_mode = Compressor::COMP_NONE;  // negotiated mode
+  uint32_t con_method = Compressor::COMP_ALG_NONE; // negotiated method
+
+  bool is_compress() const { return con_mode != Compressor::COMP_NONE; }
+
+  uint32_t get_method() const { return con_method; }
+
+  uint32_t get_mode() const { return con_mode; }
```

Comment:
There seems to be some (harmless) confusion between sint and int as the underlying type of the enums. Which leads to the question: why not use strong enums here, instead of integer tags?

Reply:
i concur with @ronen-fr . might want to use scoped enum here.